src/helpers.js

```diff
@@ -870,6 +870,32 @@ export function getNodeDevSource(dev) {
         const bus = dev.busnum;
 
         source = `${bus}.${device}`;
+    } else {
+        throw new Error(`getNodeDevSource: unsupport device type '${dev.type}'`);
+    }
+
+    return source;
+}
+
+export function getHostDevSourceObject(dev) {
+    let source;
+
+    if (dev.type === "pci") {
+        const domain = dev.source.address.domain;
+        const bus = dev.source.address.bus;
+        const slot = dev.source.address.slot;
+        const func = dev.source.address.func;
+
+        source = { domain, bus, slot, func };
+    } else if (dev.type === "usb") {
+        const device = dev.source.device;
+        const bus = dev.source.bus;
+        const vendor = dev.source.vendor.id;
+        const product = dev.source.product.id;
+
+        source = { vendor, product, bus, device };
+    } else {
+        throw new Error(`getHostDevSourceObject: unsupport device type '${dev.type}'`);
```

Comment:
@yunmingyang I squashed your commit about throwing this error message to already existing commit. Also changed the error message to also show which type is not supported.

Reply:
OK, thanks